scalafix/input/src/main/scala/fix/SetMapSrc.scala

```diff
@@ -3,10 +3,22 @@ rule = "scala:fix.Scalacollectioncompat_newcollections"
  */
 package fix
 
-class SetMapSrc(set: Set[Int], map: Map[Int, Int]) {
-  set + (2, 3)
-  map + (2 -> 3, 3 -> 4)
-  (set + (2, 3)).map(x => x)
-  set + (2, 3) - 4
-  map.mapValues(_ + 1)
-}
\ No newline at end of file
+import scala.collection
+import scala.collection.immutable
+import scala.collection.mutable.{Map, Set} // Challenge to make sure the scoping is correct
+
+class SetMapSrc(iset: immutable.Set[Int], 
+                cset: collection.Set[Int],
+                imap: immutable.Map[Int, Int],
+                cmap: collection.Map[Int, Int]) {
+  iset + (2, 3)
+  imap + (2 -> 3, 3 -> 4)
+  (iset + (2, 3)).toString
+  iset + (2, 3) - 4
+  imap.mapValues(_ + 1)
+  iset + 1
+  iset - 2
+  cset + 1
+  cset - 2
+  cmap + (2 -> 3) + ((4, 5))
```

Comment:
Can you also add the following test case?

~~~ diff
- imap + (2 -> 3) + ((4, 5))
+ imap + (2 -> 3) + ((4, 5))
~~~